tests/test_estimate_params.py

```diff
@@ -0,0 +1,965 @@
+        
+        # Passing in Runs directly
+        with self.assertRaises(ValueError):
+            m.estimate_params(wrongData)
+
+        # Defining wrongOutputs to test parameter length tests.
+        # Arbitrary Outputs created here.
+        wrongOutputs = [
+            {'x': 1.83},
+            {'x': 36.95},
+            {'x': 62.36},
+            {'x': 77.81},
+            {'x': 83.45},
+            {'x': 79.28},
+            {'x': 65.3},
+            {'x': 41.51},
+        ]
+
+        # Formatting incorrect Output data.
+        wrongData = [(results.times, results.inputs, wrongOutputs)]
+
+        # Wrong outputs parameter length
+        with self.assertRaises(ValueError) as cm:
+            m.estimate_params(times=[results.times], inputs=[results.inputs], outputs=[wrongOutputs])
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 0. Length of times: 9, Length of inputs: 9, Length of outputs: 8',
+            str(cm.exception)
+        )
+
+        # Both inputs and outputs with incorrect lenghts
+        with self.assertRaises(ValueError) as cm:
+            m.estimate_params(times=[results.times], inputs=[wrongInputLen], outputs=[wrongOutputs])
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 0. Length of times: 9, Length of inputs: 8, Length of outputs: 8',
+            str(cm.exception)
+        )
+
+        # Without wrapper
+        with self.assertRaises(ValueError) as cm:
+            m.estimate_params(times=results.times, inputs=wrongInputLen, outputs= wrongOutputs)
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 0. Length of times: 9, Length of inputs: 8, Length of outputs: 8',
+            str(cm.exception)
+        )
+
+        # Length error expected, 1, 9, 1.
+        with self.assertRaises(ValueError) as cm:
+            m.estimate_params(times=[[results.times]], inputs=[results.inputs], outputs=[[results.outputs]])
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 0. Length of times: 1, Length of inputs: 9, Length of outputs: 1',
+            str(cm.exception)
+        )
+
+
+        # Passing in incorrect Runs
+        with self.assertRaises(ValueError):
+            m.estimate_params(wrongData)
+
+        m.parameters['thrower_height'] = 1.5
+        m.parameters['throwing_speed'] = 25
+
+        # Testing functionality works without having a parent wrapper
+        m.estimate_params(times=results.times, inputs=results.inputs, outputs=results.outputs)
+        for key in keys:
+            self.assertAlmostEqual(m.parameters[key], gt[key], 2)
+
+        # Differnet types of wrappers should not affect function
+        # Testing functionality works with having only a few parameters defined in wrapper sequences
```

Comment:
```suggestion
        # Testing functionality works with having only a few wrapper sequences
```
Might need rewording, but "parameters" means something different, so we don't want to use that here